### PR TITLE
[WIP] FluentReplaceRector

### PR DESCRIPTION
**Pull request by carusogabriel**

### CODE REVIEW

File: src/Rector/Dynamic/FluentReplaceRector.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        if (! $returnExpr instanceof Variable) {
+            return false;
+        }
+
+        return $returnExpr->name === 'this';
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        return null;
```
Comment (carusogabriel) [inline src/Rector/Dynamic/FluentReplaceRector.php]: @TomasVotruba Returning `null` shouldn't remove `return $this`?

Comment (TomasVotruba) [reply src/Rector/Dynamic/FluentReplaceRector.php]: It should be `Nop` instead. Null means *do nothing*

File: tests/Rector/Dynamic/FluentReplaceRector/Wrong/wrong.php.inc [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    private $someClass;
+
+    public function someFunction()
+    {
+        $this->someClass = new SomeClass();
+
+        $this->someClass->someFunction()
+                        ->otherFunction();
+    }
+}
```
Comment (TomasVotruba) [inline tests/Rector/Dynamic/FluentReplaceRector/Wrong/wrong.php.inc]: Beautiful test case btw :+1:

Comment (carusogabriel) [reply tests/Rector/Dynamic/FluentReplaceRector/Wrong/wrong.php.inc]: Thanks, I think in a good step to start.

File: src/Rector/Dynamic/FluentReplaceRector.php (hunk 1 above)
Comment (carusogabriel) [reply src/Rector/Dynamic/FluentReplaceRector.php]: I keep receiving an error: `Uninitialized string offset: 0` :disappointed:

**Comment by TomasVotruba** [comment]
Btw, now I see you'r sending this from fork. 
Since you're member, you don't have to fork, you can send PR from this repository too (at least you should be able too, let me know to fix that if not)

**Comment by carusogabriel** [comment]
No, as fair as I know, I don’t have any permissions, neither create branches, neither merge PR or close issues.

**Comment by TomasVotruba** [comment]
@carusogabriel I tried to add you to collaborator team. Could you try now?

**Comment by TomasVotruba** [comment]
Surpassed by #286

**Pull request closed without merging**